## Dictionary Maker

Starting off with [Ogden's Basic Wordlist](http://ogden.basic-english.org/words.html), adding in contractions for later (and after the fact)

In [131]:
def get_basic_english():
        '''
        Creates a list of basic english words from a csv file containing
        Ogden's 850 basic english words.
        '''
        # Convert basic english words to a list
        basic_english_df = pd.read_csv('../data/basic_english_wordlist.csv')
        print basic_english_df.head()
        basic_english = []
        basic_english = [a for a in basic_english_df['WORD']]
        # add the various conjugations of 'to be' and 'a'
        basic_english.append('an')
        basic_english.append('is')
        basic_english.append('was')
        basic_english.append('are')
        basic_english.append('were')
        basic_english.append('they')
        # 'I' causes weird issues...
        basic_english[basic_english.index('I')] = 'big'
        basic_english.append('she')
        basic_english.append('hers')
        basic_english.append('his')
        basic_english.append('my')
        basic_english.append('him')
        basic_english.append('her')
        basic_english.append('your')
        basic_english.append('their')
        basic_english.append('might')
        basic_english.append('must')
        basic_english.append('can')
        basic_english.append('did')
        basic_english.append('could')
        basic_english.append('should')
        basic_english.append('would')
        basic_english.append('that')
        basic_english.append('what')
        basic_english.append('we')
        basic_english.append('small')
        basic_english[basic_english.index('colour')] = 'color'
        print 'basic_english has {} words'.format(len(basic_english))
        return basic_english

In [132]:
import pandas as pd
basic_english = get_basic_english()

      WORD
0        a
1     able
2    about
3  account
4     acid
basic_english has 875 words


So, 875 is not a lot of words, and even CK Ogden noted that each industry will need around 150 extra words to communicate in that industry.  So, first things first, let's grab the [internaion words](http://ogden.basic-english.org/intlword.html) that the Ogden crowd already likes...

In [133]:
new_words = 'international, '
# International Nouns
new_words = new_words + 'alcohol, aluminum, automobile, bank, bar, beef, beer, calendar, check, chemist, chocolate, chorus, cigarette, club, coffee, colony, dance, engineer, gas, hotel, influenza, lava, madam, nickel, opera, orchestra, paraffin, park, passport, patent, phonograph, piano, police, post, program, propaganda, radio, restaurant, sir, sport, taxi, tea, telegram, telephone, terrace, theater, tobacco, university, whisky, zinc, '
# Names of sciences
new_words = new_words + 'Algebra, Arithmetic, Biology, Chemistry, Geography, Geology, Geometry, Mathematics, Physics, Physiology, Psychology, Zoology, '
#International names used in titles, organizations, diplomacy, etc...
new_words = new_words + 'College, Dominion, Embassy, Empire, Imperial, King, Museum, President, Prince, Princess, Queen, Royal, '
# general utility...
new_words = new_words + 'ammonia, asbestos, autobus, ballet, cafe, catarrh, champagne, chauffeur, circus, citron, cocktail, cognac, dynamite, encyclopedia, glycerin, hyena, hygiene, hysteria, inferno, jazz, liqueur, macaroni, malaria, mania, nicotine, olive, omelet, opium, paradise, penguin, platinum, potash, pajamas, pyramid, quinine, radium, referendum, rheumatism, rum, salad, sardine, tapioca, toast, torpedo, vanilla, violin, visa, vodka, volt, zebra, '
# Onomatopoeic (new word just added to my vocab right there)
new_words = new_words + 'bang, hiss, meow, purr, quack, bark, woof'  # last two aded by me, dogs ftw
# time and numbers
new_words = new_words + 'January, February, March, April, May, June, July, August, September, October, November, December, '
new_words = new_words + 'Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday, '
new_words = new_words + 'One, two, three, four, five, six, seven, eight, nine, ten, eleven, twelve, thirteen, fourteen, fifteen, sixteen, twenty, twenty-one, thirty, forty, fifty, hundred, thousand, million, once, twice, half, quarter, third, fourth, fifth, '
# New century skipped for now, add later

#the science list
new_words = new_words + 'active, adjacent, adsorption, age, alternative, application, arc, area, arrangement, ash, axis, break, bubble, capacity, case, cell, column, component, compound, continuous, cross, decrease, deficiency, deposit, determining, difference, difficulty, direct, disappearance, discharge, disturbance, elimination, environment, equation, evaporation, exact, experiment, explanation, focus, friction, fusion, generation, groove, guard, hinge, impurity, individual, interpretation, investigation, joint, latitude, layer, length, link, longitude, mean, melt, mixture, nucleus, origin, path, per, pressure, projection, proof, reference, relative, reproduction, resistance, rigidity, rock, rot, rotation, screen, seal, section, sensitivity, shadow, shear, shell, similarity, solution, spark, specialization, specimen, stimulus, strain, strength, stress, substitution, successive, supply, surface, swelling, thickness, thrust, tide, transmission, transparent, tube, valve, '
new_words = new_words + 'anesthetic, bomb, electricity, magnetic, microscope, neutron, organism, petroleum, serum, thermometer, tropism, turbine, vitamin, '
new_words = new_words + 'allotropy, endothelium, hemocyanin, isomorphism, phylogeny, tropism, '
# phys sci
new_words = new_words + 'absorption, beaker, buoyancy, charge, circuit, clip, coil, collision, combination, conductor, conservation, corrosion, density, dilution, dissipation, explosion, flask, fume, funnel, furnace, grating, image, insulator, lag, lens, medium, oxidation, particle, pendulum, plug, porcelain, radiation, reagent, receiver, reflux, repulsion, residue, reversible, saturated, solvent, stable, stream, suspension, switch, tongs, trap, valency, vapor, vortex, wedge, '
# geology
new_words = new_words + 'accessory, birefringence, cast, cave, clay, cleavage, contour, desert, dike, dip, drift, erosion, eruption, extinction, fan, fault, flint, flood, flow, foliation, fracture. glacier, gravel, ground, hill, igneous, inclusion, intercept, interpenetration, intrusion, lake, limestone, mud, ore, outcrop, outlier, overlap, plain, scarp, schist, sedimentary, shale, shore, sill, slate, strike, texture, twin, unconformity, valley, '
# math & mechanics
new_words = new_words + 'acceleration, amplitude, approximation, arbitrary, circumference, congruent, constant, cusp, damping, denominator, diameter, divisor, fraction, infinity, integer, intersection, inverse, lever, load, locus, magnitude, momentum, multiple, multiplication, node, numerator, oblique, piston, prime, probability, product, projectile, pulley, quantity, quotient, ratio, reciprocal, rectangle, recurring, reinforcement, resolution, subtraction, sum, term, total, triangle, unknown, variable, velocity, volume, '
# biology
new_words = new_words + 'abdomen, appendage, bark, beak, bud, cartilage, cavity, claw, climber, creeper, degenerate, domesticating, duct, ferment, fertilizing, fiber, fin, fresh, germinating, gill, gland, hoof, host, inheritance, jaw, juice, kidney, liver, lung, mature, metabolism, parent, petal, pollen, sac, scale, secretion, sepal, skull, slide, soil, stain, stalk, stamen, sucker, thorax, tissue, vascular, vestigial, wild, '
# commerece
new_words = new_words + 'acceptance, address, agency, allowance, assets, assistant, average, bale, bankrupt, barrel, bill, broker, certificate, charge, claim, client, code, complaint, consignment, cost, court, customs, debit, deck, defect, delivery, export, gross, guarantee, hire, hold, import, investment, liability, license, load, loan, packing, pair, partner, purchase, reference, retail, sale, sample, show, sight, strike, supply, wholesale, '
# econ
new_words = new_words + 'accident, arbitration, asset, average, bill, broker, budget, circulation, combine, consumer, conversion, correlation, cost, deflation, demand, deposit, discount, efficiency, effort, employer, experiment, factor, fatigue, guarantee, habit, index, inflation, investment, liability, loan, margin, monopoly, partner, pension, plan, population, purchase, rent, retail, sale, saving, security, service, share, speculation, statistics, stimulus, strike, supply, wholesale, '
# buis and econ
new_words = new_words + 'asset, average, bill, broker, cost, guarantee, investment, liability, loan, partner, purchase, retail, sale, strike, supply, wholesale, '
new_words = new_words + 'capital, firm, wages, '
# Trade, despite the greyed-outness of the text
new_words = new_words + 'cash, premium, service, slump, dock, net, warrant, '
new_words = new_words + 'canal, capital, carton, consul, guarantee, minus, percent, plus, propaganda, stand, wagon, '
new_words = new_words + 'balance, forward, case, royalty, duties, '

# Verse
#coincidently handles some words I was worried about for Dumas, like sword
# God/god1 thing -> god, since capitalization would be maintained thru other bits of code.
new_words = new_words + 'angel, arrow, beast, blind, bow, breast, bride, brow, bud, calm, child, cross, crown, curse, dawn, delight, dew, dove, dream, eagle, eternal, evening, evil, fair, faith, fate, feast, flock, flow, fountain, fox, gentle, glad, glory, god, grace, grape, grief, guest, hawk, heaven, hell, hill, holy, honey, honor, image, ivory, joy, lamb, lark, life, lion, lord, meadow, melody, mercy, noble, passion, perfume, pity, pool, praise, prayer, pride, priest, purple, rapture, raven, robe, rock, rose, rush, search, shining, shower, sorrow, soul, spear, spirit, storm, stream, strength, sword, thief, tower, travel, valley, veil, vine, violet, virgin, virtue, vision, wandering, wealth, weariness, weeping, wisdom, wolf, wonder, '

# Bible
new_words = new_words + 'altar, ark, ass, ax, baptism, blessing, captain, cattle, circumcision, cock, deceit, disciple, envy, flesh, forgiveness, generation, herd, heritage, husband, kingdom, leaven, leper, locust, master, neighbor, oath, ox, people, pillar, preaching, prophet, revelation, righteousness, saint, salvation, savior, scribe, sin, spice, tent, testament, thorn, tribe, virgin, widow, wife, witness, world, worship, wrath, yoke, '

# Social Science... funny, since making up new words is their thing
new_words = new_words = 'heredity, incentive, integration, standardization, State, technology, unemployment, environment, association, attitudes, class, community, culture, custom, '
new_words = new_words = 'capital, firm, wages '


new_words = new_words = 'decentralization, structure, concentration, control, will, '


# Net words...
# note, many repeats, I will be 'set'ing these later, so i include all to make sure I don't miss one
new_words = new_words + 'alphabet, back , bar , base , board , body , boot , box , broken , bulletin , button , cable , card , check , clear , center , connection , control , copy , cord , current , cut , down , dictionary , division , drop , end , engine , field , fixed , folder , forward , group , hard , head , help, in , ink , join , key , knowledge , language , last , left , library , line , list , lock , log, machine , map , memo , memory , move , name , net , news , observation , off , office , on , open , out , page , parallel, park , person , pin , pointer , post , print , program , process , public , pull , push , put , range , reading , record , right , roll , run , send , shut , size , smooth, soft , sort , sound , space , start , statement , step , sticky , store , support , syntax , system , table , test , text , thread , tray , unit , up , value , view , voice , window , wire , word , work, '
new_words = new_words + 'backspace , bookmark, input , keyboard , lockout , lockup, lookup, online , output , network , readme , shutdown , startup , update, '
# these were two-word expressions, want to make sure I get both words seperatly and together
new_words = new_words + 'back, button, black, box, cold, start, high, memory, news, group, number, lock, page, up, page, down, power, cord, space, bar, support, line, thumb, drive, '

# including candidates
new_words = new_words + ' analog , archive , base , center , client , column , computer , control , copy , cyberspace , disk , diskette , division , hotlink , icon , information , internet , list , logic , memory , menu , modem , page , parallel , personal , personal computer , PC , post , program , project , protocol , server , system , table , test , text , type , variable , virtual , website, '
new_words = new_words + 'code , connection , data , digital , domain , homepage , insert , monitor , packet , park , value , webmaster, '
new_words = new_words + 'base, center, copy, list, page, post, program, project, system, table, test, '


# even though NOT basic words, adding them anyway
new_words = new_words + 'address, align, analog, center, click, code, column, computer, cyber, data, delete, digital, director, disk, domain, drag, file, font, hold, home, hyper, information, insert, internet, link, load, logic, mail, menu, modem, monitor, mouse, packet, protocol, recent, reply, row, search, serial, site, text, type, web, '

stuff = 'absence , accident [b] , address [b] , adventure , advice , age[s] , agent , ago , along , also , always , ambition , anchor , ankle , arrangements [s], ash [s] , awkward , balcony , barrel [b] , beard , beat , behind , belt , bet , blame , blanket , both , bottom , brave , breakfast , breast [v] , bubble [s] , bud [v] , bunch , burial , busy , calculation , call , capital [g] , carpet , case [s] , ceremony , chair , character , child [v] , chimney , china , choice , civilization , clay , clever , collection , column [s,c] , communications , concept , concrete , convenient , cool , corner , cost [b] , crop , cross[s,v] , cunning , date , demand [b] , department , dew [v] , difficulty [s] , dinner , disgrace , dislike , ditch , dive , divorce , doll , dreadful , dream [v] , dull , duty , each , easy , economy , effort [b] , either , employer [b] , empty , enemy , envelope , envy , evening [v] , exact [s] , excitement , exercise , explanation [s] , explosion , expression , eyebrow , eyelash , failure , fair [v] , famous , fan , fastening , fever , figure , financial , flash , flesh , flood , flour , forecast , forehead , foreign , forgiveness , fresh , frost , frozen , funny , fur , furniture , gate , generation [s] , germ , god [v] , grand , grateful , grease , grief [v] , grocery , ground , guard [s] , guess , gum , habit [b] , handkerchief , handle , heavy , hedge , hil l[v] , holiday , home , honest , honey [v] , human , hunt , hurry , hurt , husband , imagination , innocent , institution , intelligent , invitation , jam , jaw , jealous , jerk , joint [s] , jug , juice , jury , justice , kennel , kidney , kitchen , knock , lace , lame , lamp , large , lawyer , lazy , lecture , legal , lesson , lever , license , lid , life [v] , lime , liver , load [b] , local , loop , luck , lump , lunch , lung , magic , manner , many , marble , marriage , mast , mattress , mean [s] , meaning , medicine , member , mess , message , mill , mineral , model , modern , modest , mood , moral , moustache , mud , murder , nasty , nature , navy , neat , neglect , neighbor , nest , next , nice , nostril , nurse , obedient , officer , orchestra , organ , origin [s] , oval , own , pad , pair [b] , pan , paragraph , party , passage , patience , pedal , people , perfect , plaster , poetry , pool [v] , practice , praise [v] , prayer [v] , prick , priest [v] , progress , promise , proof [s] , proud [v] , pupil , pure , race , real , reference [s,t] , remark , remedy , repair , result , revenge , rich , rise , rival , rude , rust , sale [b] , satisfaction , saucer , scratch , screen [s] , search , selfish , sentence , set , shave , sheet , shell [s] , shoulder , show [b] , sight [b], since , sleeve , social , soldier , sorry , spark [s] , spirit [v] , spit , splash , spot , stable, stair, steady , storm [v] , strain , straw , stream [v] , string , study , subject , success , sum , supply [s,b] , surface [s] , surgeon , suspicious , swing , sympathetic , tailor , tame , tap , tear , tent , thief [v] , thimble , threat , tie , too , towel , tower [v] , traffic , tragedy , trap , travel [v] , treatment , truck , tube [s], tune , tunnel , twin , typist , ugly , understanding , universe , valle y[v] , vegetable , victim , victory , vote , weak , welcome , whether , widow , wife , wild , world , wreck , wrist , yawn'
new_words = new_words + stuff.replace(' [b]','').replace('[s]','').replace('[s,c]','').replace('[g]','').replace('[v]','').replace('[s,b]','').replace('[s,t]','').replace(' ,',',')+', '

# doubles...
new_words = new_words + 'accident , age , agent , arrangement , ash , barrel , breast , bubble , bud , capital , case , child , client , column , cost , cross , demand , deposit , dew , difficulty , dream , effort , employer , evening , exact , experiment , explanation, fair , god , grief , guard , habit , hill , honey , joint , life , load , mean , origin , pool, praise , prayer, priest , proof , proud , reference , rock , sale , screen , search , shell , show , sight , spark , spirit , storm , stream , strength , supply , surface , thief , tower , travel , tube , valley , '
new_words = new_words + 'can' # sure, I'll bite
new_words.replace('trousers','pants') # 'Merica!

# Units!
new_words = new_words + 'gram, second, kelvin, ampere, mole, candela, meter, mile, inch, foot, yard, celsius, fahrenheit, centigrade, pound, stone, centimeters, kilogram, ' 

expanded_english = new_words.split(', ')
basic_english = basic_english + list(set(expanded_english))
basic_english.append('club')
basic_english.append('pistol')
print basic_english[:10]
basic_english.sort()
basic_english = list(set([a.replace(' ','').lower() for a in basic_english]))
print "Now we have {} words".format(len(basic_english))



['a', 'able', 'about', 'account', 'acid', 'across', 'act', 'addition', 'adjustment', 'advertisement']
Now we have 1331 words


Include the [most common](https://en.wikipedia.org/wiki/Most_common_words_in_English) words...


In [134]:
add = "time person year way day thing man world life hand part child eye woman place work week case point government company \
number group problem fact be have do say get make go know take see come think look want give use find tell ask work seem feel \
try leave call good new first last long great little own other old right big high different small large next early young \
important few public bad same able to of in for on with at by from up about into over after beneath under above the and \
a that I it not he as you this but his they her she or an will my one all would there their"
add_lst = add.split(' ')
basic_english =  basic_english + add_lst
basic_english.sort()
basic_english = list(set([a.replace(' ','').lower() for a in basic_english]))
print "Now we have {} words".format(len(basic_english))

Now we have 1348 words


In [121]:
import gensim
model = gensim.models.KeyedVectors.load_word2vec_format('../model/GoogleNews-vectors-negative300.bin', binary=True)

KeyboardInterrupt: 

In [141]:
import nltk.data
from nltk import pos_tag, word_tokenize
from nltk.stem.lancaster import LancasterStemmer
import cPickle as pickle
def make_dictionary(basic_english):
        '''
        This means the data file associated with this particular instance does
        not have a 'data/basic_english.pickle' file associated with it, and
        we need to make one.
        '''
        print 'Running...'
        vocab_google = model.vocab.keys()

        print len(basic_english)
        # adding contractions...
        contractions_df = pd.read_csv('../data/contractions.csv', sep=' -')
        contractions = [word for word in contractions_df['from']]
        contractions[18] = "mightn't"
        st = LancasterStemmer()
        stem_gn = [st.stem(key) for key in model.vocab.keys()]
        stem_se = [st.stem(word) for word in basic_english]
        try:
            my_dict = pickle.load(open('../data/basic_english.pickle', 'rb'))
        except:
            my_dict = {}
        stems = []
        threshold = 0.2  # much smaller and it grabs weird things
        print 'starting...'
        for sim_in in xrange(len(basic_english)):
            indices = []
            check = []
            if (len(basic_english[sim_in]) >= 2) and (stem_se[sim_in] not in stems):
                stems.append(stem_se[sim_in])
                indices = [i for i, s in enumerate(stem_gn)
                           if stem_se[sim_in] == s]
                check = [i for i, s in enumerate(vocab_google)
                         if basic_english[sim_in] == s]
                print '{}: {} of {}'.format(basic_english[sim_in], sim_in, len(basic_english))
                # check, indices
                if len(check) > 0:
                    for index in indices:
                        a = threshold
                        if (model.similarity(basic_english[sim_in], vocab_google[index]) >= a):
                            my_dict[vocab_google[index].lower()] = \
                                    [vocab_google[index].lower(),
                                     basic_english[sim_in].lower(),
                                     pos_tag([vocab_google[index].lower()])[0][1]]
                            print "    {} -> {}".format(vocab_google[index].lower(), my_dict[vocab_google[index].lower()])
                        # add itself last... to overwrite issues w/above
                    a = basic_english[sim_in].lower()
                    b = pos_tag([a])[0][1]
                    my_dict[basic_english[sim_in].lower()] = [a, a, b]
            else:
                a = basic_english[sim_in].lower()
                b = pos_tag([a.lower()])[0][1]
                my_dict[basic_english[sim_in].lower()] = [a, a, b]
                print '{}: {} of {}'.format(basic_english[sim_in], sim_in, len(basic_english))
                
            # Save it...
            with open('../data/basic_english.pickle', 'wb') as handle:
                pickle.dump(my_dict, handle, protocol=pickle.HIGHEST_PROTOCOL)
        return my_dict

In [146]:
keys = my_dict.keys()
new = []
for word in basic_english:
    if word not in keys:
        if len(word) > 0:
            new.append(word)
print len(new)

473


In [149]:
basic_english.sort()
my_dict = make_dictionary(new[1:len(new)])

Running...
472


C:\Users\darin\Anaconda2\lib\site-packages\ipykernel_launcher.py:16: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.
  app.launch_new_instance()


starting...
above: 0 of 472
    above -> [u'above', 'above', 'IN']
    above -> [u'above', 'above', 'IN']
    above -> [u'above', 'above', 'IN']
    abover -> [u'abover', 'above', 'NN']
above
absence: 1 of 472
    absents -> [u'absents', 'absence', 'NNS']
    absented -> [u'absented', 'absence', 'VBN']
    absentee -> [u'absentee', 'absence', 'NN']
    absenting -> [u'absenting', 'absence', 'VBG']
    absenteeism -> [u'absenteeism', 'absence', 'NN']
    absentees -> [u'absentees', 'absence', 'NNS']
    absence -> [u'absence', 'absence', 'NN']
    absent -> [u'absent', 'absence', 'NN']
    absence -> [u'absence', 'absence', 'NN']
    absent -> [u'absent', 'absence', 'NN']
    absences -> [u'absences', 'absence', 'NNS']
    absentees -> [u'absentees', 'absence', 'NNS']
    absence -> [u'absence', 'absence', 'NN']
    absences -> [u'absences', 'absence', 'NNS']
absence
accident: 2 of 472
    accidental -> [u'accidental', 'accident', 'NN']
    accident -> [u'accident', 'accident', 'NN']
  

archive: 20 of 472
    archion -> [u'archion', 'archive', 'NN']
    archival -> [u'archival', 'archive', 'NN']
    archive -> [u'archive', 'archive', 'JJ']
    archiving -> [u'archiving', 'archive', 'VBG']
    archive -> [u'archive', 'archive', 'JJ']
    archive -> [u'archive', 'archive', 'JJ']
    archivist -> [u'archivist', 'archive', 'NN']
    archival -> [u'archival', 'archive', 'NN']
    archiving -> [u'archiving', 'archive', 'VBG']
    archivists -> [u'archivists', 'archive', 'NNS']
    archived -> [u'archived', 'archive', 'VBN']
    archiver -> [u'archiver', 'archive', 'NN']
    archives -> [u'archives', 'archive', 'NNS']
    archived -> [u'archived', 'archive', 'VBN']
    archiver -> [u'archiver', 'archive', 'NN']
    archives -> [u'archives', 'archive', 'NNS']
    archivists -> [u'archivists', 'archive', 'NNS']
    archivally -> [u'archivally', 'archive', 'RB']
    archiv -> [u'archiv', 'archive', 'NN']
    archivist -> [u'archivist', 'archive', 'NN']
    archiving -> [u'archi

bookmark: 39 of 472
    bookmark -> [u'bookmark', 'bookmark', 'NN']
    bookmarking -> [u'bookmarking', 'bookmark', 'NN']
    bookmarking -> [u'bookmarking', 'bookmark', 'NN']
    bookmarked -> [u'bookmarked', 'bookmark', 'VBN']
    bookmarks -> [u'bookmarks', 'bookmark', 'NNS']
    bookmarks -> [u'bookmarks', 'bookmark', 'NNS']
    bookmarkers -> [u'bookmarkers', 'bookmark', 'NNS']
    bookmarked -> [u'bookmarked', 'bookmark', 'VBN']
    bookmark -> [u'bookmark', 'bookmark', 'NN']
    bookmarks -> [u'bookmarks', 'bookmark', 'NNS']
bookmark
both: 40 of 472
    both -> [u'both', 'both', 'DT']
    bothe -> [u'bothe', 'both', 'NN']
    boths -> [u'boths', 'both', 'NNS']
    both -> [u'both', 'both', 'DT']
    both -> [u'both', 'both', 'DT']
both
bottom: 41 of 472
    bottom -> [u'bottom', 'bottom', 'NN']
    bottoming -> [u'bottoming', 'bottom', 'NN']
    bottome -> [u'bottome', 'bottom', 'NN']
    bottoms -> [u'bottoms', 'bottom', 'NNS']
    bottom -> [u'bottom', 'bottom', 'NN']
    bott

carpet: 57 of 472
    carpet -> [u'carpet', 'carpet', 'NN']
    carpet -> [u'carpet', 'carpet', 'NN']
    carpeted -> [u'carpeted', 'carpet', 'VBN']
    carpets -> [u'carpets', 'carpet', 'NNS']
    carpeted -> [u'carpeted', 'carpet', 'VBN']
    carpets -> [u'carpets', 'carpet', 'NNS']
    carpet -> [u'carpet', 'carpet', 'NN']
    carpeting -> [u'carpeting', 'carpet', 'NN']
    carpeting -> [u'carpeting', 'carpet', 'NN']
    carpets -> [u'carpets', 'carpet', 'NNS']
carpet
case: 58 of 472
    case -> [u'case', 'case', 'NN']
    cases -> [u'cases', 'case', 'NNS']
    cases -> [u'cases', 'case', 'NNS']
    case -> [u'case', 'case', 'NN']
    cas -> [u'cas', 'case', 'NN']
    cases -> [u'cases', 'case', 'NNS']
case
celsius: 59 of 472
    celsius -> [u'celsius', 'celsius', 'NN']
    celsius -> [u'celsius', 'celsius', 'NN']
    celsius -> [u'celsius', 'celsius', 'NN']
celsius
center: 60 of 472
    center -> [u'center', 'center', 'NN']
    centralizers -> [u'centralizers', 'center', 'NNS']
   

click: 74 of 472
    clicked -> [u'clicked', 'click', 'NNS']
    clicker -> [u'clicker', 'click', 'NN']
    clicks -> [u'clicks', 'click', 'NNS']
    clickers -> [u'clickers', 'click', 'NNS']
    clicking -> [u'clicking', 'click', 'VBG']
    clickable -> [u'clickable', 'click', 'JJ']
    clicks -> [u'clicks', 'click', 'NNS']
    clickity -> [u'clickity', 'click', 'NN']
    click -> [u'click', 'click', 'NN']
    clickety -> [u'clickety', 'click', 'NN']
    clicks -> [u'clicks', 'click', 'NNS']
    clickers -> [u'clickers', 'click', 'NNS']
    clicking -> [u'clicking', 'click', 'VBG']
    click -> [u'click', 'click', 'NN']
    clickables -> [u'clickables', 'click', 'NNS']
    clickable -> [u'clickable', 'click', 'JJ']
    clickety -> [u'clickety', 'click', 'NN']
    clicked -> [u'clicked', 'click', 'NNS']
    clicker -> [u'clicker', 'click', 'NN']
    clicking -> [u'clicking', 'click', 'VBG']
    click -> [u'click', 'click', 'NN']
click
client: 75 of 472
    cliental -> [u'cliental', 'cl

concentration: 82 of 472
    concede -> [u'concede', 'concentration', 'NN']
    concentrate -> [u'concentrate', 'concentration', 'NN']
    concentrating -> [u'concentrating', 'concentration', 'VBG']
    conciousness -> [u'conciousness', 'concentration', 'NN']
    concentration -> [u'concentration', 'concentration', 'NN']
    concentrating -> [u'concentrating', 'concentration', 'VBG']
    conciseness -> [u'conciseness', 'concentration', 'NN']
    concentrate -> [u'concentrate', 'concentration', 'NN']
    concentrated -> [u'concentrated', 'concentration', 'VBN']
    concentrations -> [u'concentrations', 'concentration', 'NNS']
    concentricity -> [u'concentricity', 'concentration', 'NN']
    concentrations -> [u'concentrations', 'concentration', 'NNS']
    concentration -> [u'concentration', 'concentration', 'NN']
    conceding -> [u'conceding', 'concentration', 'VBG']
    concentrates -> [u'concentrates', 'concentration', 'NNS']
    contentedness -> [u'contentedness', 'concentration', 

cunning: 92 of 472
    cunis -> [u'cunis', 'cunning', 'NN']
    cunning -> [u'cunning', 'cunning', 'VBG']
    cunning -> [u'cunning', 'cunning', 'VBG']
    cunningly -> [u'cunningly', 'cunning', 'RB']
    cunning -> [u'cunning', 'cunning', 'VBG']
    cunningness -> [u'cunningness', 'cunning', 'NN']
    cunth -> [u'cunth', 'cunning', 'NN']
    cunningly -> [u'cunningly', 'cunning', 'RB']
cunning
cut: 93 of 472
    cuts -> [u'cuts', 'cut', 'NNS']
    cut -> [u'cut', 'cut', 'NN']
    cuts -> [u'cuts', 'cut', 'NNS']
    cuttable -> [u'cuttable', 'cut', 'JJ']
    cutting -> [u'cutting', 'cut', 'VBG']
    cutting -> [u'cutting', 'cut', 'VBG']
    cutting -> [u'cutting', 'cut', 'VBG']
    cutt -> [u'cutt', 'cut', 'NN']
    cuts -> [u'cuts', 'cut', 'NNS']
    cutters -> [u'cutters', 'cut', 'NNS']
    cuting -> [u'cuting', 'cut', 'VBG']
    cut -> [u'cut', 'cut', 'NN']
    cutted -> [u'cutted', 'cut', 'VBN']
    cutter -> [u'cutter', 'cut', 'NN']
    cut -> [u'cut', 'cut', 'NN']
cut
cyber: 94 o

dinner: 107 of 472
    dine -> [u'dine', 'dinner', 'NN']
    dinner -> [u'dinner', 'dinner', 'NN']
    dinners -> [u'dinners', 'dinner', 'NNS']
    dinners -> [u'dinners', 'dinner', 'NNS']
    dinning -> [u'dinning', 'dinner', 'VBG']
    diner -> [u'diner', 'dinner', 'NN']
    dine -> [u'dine', 'dinner', 'NN']
    dining -> [u'dining', 'dinner', 'VBG']
    dined -> [u'dined', 'dinner', 'VBN']
    diners -> [u'diners', 'dinner', 'NNS']
    dinners -> [u'dinners', 'dinner', 'NNS']
    dining -> [u'dining', 'dinner', 'VBG']
    dine -> [u'dine', 'dinner', 'NN']
    dinner -> [u'dinner', 'dinner', 'NN']
    diners -> [u'diners', 'dinner', 'NNS']
    dinned -> [u'dinned', 'dinner', 'VBN']
    dinner -> [u'dinner', 'dinner', 'NN']
    dines -> [u'dines', 'dinner', 'NNS']
    diner -> [u'diner', 'dinner', 'NN']
    dined -> [u'dined', 'dinner', 'VBN']
    diners -> [u'diners', 'dinner', 'NNS']
    dining -> [u'dining', 'dinner', 'VBG']
dinner
director: 108 of 472
    directorship -> [u'direct

dull: 121 of 472
    dulling -> [u'dulling', 'dull', 'VBG']
    dull -> [u'dull', 'dull', 'NN']
    duller -> [u'duller', 'dull', 'NN']
    dull -> [u'dull', 'dull', 'NN']
    dullness -> [u'dullness', 'dull', 'NN']
    dulls -> [u'dulls', 'dull', 'NNS']
    dullish -> [u'dullish', 'dull', 'NN']
    duller -> [u'duller', 'dull', 'NN']
    dullness -> [u'dullness', 'dull', 'NN']
    dully -> [u'dully', 'dull', 'RB']
    dull -> [u'dull', 'dull', 'NN']
    dulled -> [u'dulled', 'dull', 'VBN']
dull
duty: 122 of 472
    duty -> [u'duty', 'duty', 'NN']
    dutiable -> [u'dutiable', 'duty', 'JJ']
    dutifully -> [u'dutifully', 'duty', 'RB']
    duty -> [u'duty', 'duty', 'NN']
    duties -> [u'duties', 'duty', 'NNS']
    dutiful -> [u'dutiful', 'duty', 'NN']
    duty -> [u'duty', 'duty', 'NN']
    duties -> [u'duties', 'duty', 'NNS']
duty
each: 123 of 472
    each -> [u'each', 'each', 'DT']
    each -> [u'each', 'each', 'DT']
    each -> [u'each', 'each', 'DT']
each
easy: 124 of 472
    easi

experiment: 136 of 472
    experimental -> [u'experimental', 'experiment', 'NN']
    experiment -> [u'experiment', 'experiment', 'NN']
    experimented -> [u'experimented', 'experiment', 'VBN']
    experimentation -> [u'experimentation', 'experiment', 'NN']
    experimenting -> [u'experimenting', 'experiment', 'VBG']
    experimentally -> [u'experimentally', 'experiment', 'RB']
    experimenters -> [u'experimenters', 'experiment', 'NNS']
    experimenter -> [u'experimenter', 'experiment', 'NN']
    experiment -> [u'experiment', 'experiment', 'NN']
    experiments -> [u'experiments', 'experiment', 'NNS']
    experimenter -> [u'experimenter', 'experiment', 'NN']
    experimentally -> [u'experimentally', 'experiment', 'RB']
    experimenters -> [u'experimenters', 'experiment', 'NNS']
    experimentalist -> [u'experimentalist', 'experiment', 'NN']
    experimentation -> [u'experimentation', 'experiment', 'NN']
    experimentalists -> [u'experimentalists', 'experiment', 'NNS']
    experimen

fastening: 147 of 472
    fastened -> [u'fastened', 'fastening', 'VBN']
    fastener -> [u'fastener', 'fastening', 'NN']
    fastenings -> [u'fastenings', 'fastening', 'NNS']
    fastens -> [u'fastens', 'fastening', 'NNS']
    fastening -> [u'fastening', 'fastening', 'VBG']
    fastened -> [u'fastened', 'fastening', 'VBN']
    fastener -> [u'fastener', 'fastening', 'NN']
    fastening -> [u'fastening', 'fastening', 'VBG']
    fasteners -> [u'fasteners', 'fastening', 'NNS']
    fasteners -> [u'fasteners', 'fastening', 'NNS']
    fasten -> [u'fasten', 'fastening', 'NNS']
    fasten -> [u'fasten', 'fastening', 'NNS']
fastening
fever: 148 of 472
    feverous -> [u'feverous', 'fever', 'JJ']
    fever -> [u'fever', 'fever', 'NN']
    fevers -> [u'fevers', 'fever', 'NNS']
    fevered -> [u'fevered', 'fever', 'VBN']
    fever -> [u'fever', 'fever', 'NN']
    fevers -> [u'fevers', 'fever', 'NNS']
    fever -> [u'fever', 'fever', 'NN']
    feverish -> [u'feverish', 'fever', 'JJ']
    feverish ->

forehead: 161 of 472
    foreheads -> [u'foreheads', 'forehead', 'NNS']
    foreheaded -> [u'foreheaded', 'forehead', 'VBN']
    forehead -> [u'forehead', 'forehead', 'NN']
    forehead -> [u'forehead', 'forehead', 'NN']
forehead
foreign: 162 of 472
    foreigners -> [u'foreigners', 'foreign', 'NNS']
    foreign -> [u'foreign', 'foreign', 'JJ']
    foreignness -> [u'foreignness', 'foreign', 'NN']
    foreigner -> [u'foreigner', 'foreign', 'NN']
    foreigners -> [u'foreigners', 'foreign', 'NNS']
    foreign -> [u'foreign', 'foreign', 'JJ']
    foreigners -> [u'foreigners', 'foreign', 'NNS']
    foreign -> [u'foreign', 'foreign', 'JJ']
foreign
forgiveness: 163 of 472
    forgiveable -> [u'forgiveable', 'forgiveness', 'JJ']
    forgiveness -> [u'forgiveness', 'forgiveness', 'NN']
    forgiveness -> [u'forgiveness', 'forgiveness', 'NN']
    forgivable -> [u'forgivable', 'forgiveness', 'JJ']
    forgiveness -> [u'forgiveness', 'forgiveness', 'NN']
    forgivable -> [u'forgivable', 'forgive

grocery: 179 of 472
    grocery -> [u'grocery', 'grocery', 'NN']
    groceries -> [u'groceries', 'grocery', 'NNS']
    groceries -> [u'groceries', 'grocery', 'NNS']
    grocery -> [u'grocery', 'grocery', 'NN']
    groceries -> [u'groceries', 'grocery', 'NNS']
    grocery -> [u'grocery', 'grocery', 'NN']
grocery
ground: 180 of 472
    ground -> [u'ground', 'ground', 'NN']
    ground -> [u'ground', 'ground', 'NN']
    grounded -> [u'grounded', 'ground', 'VBN']
    grounders -> [u'grounders', 'ground', 'NNS']
    grounding -> [u'grounding', 'ground', 'VBG']
    ground -> [u'ground', 'ground', 'NN']
    grounds -> [u'grounds', 'ground', 'NNS']
    grounds -> [u'grounds', 'ground', 'NNS']
ground
guard: 181 of 472
    guarding -> [u'guarding', 'guard', 'VBG']
    guards -> [u'guards', 'guard', 'NNS']
    guards -> [u'guards', 'guard', 'NNS']
    guarded -> [u'guarded', 'guard', 'VBN']
    guard -> [u'guard', 'guard', 'NN']
    guarded -> [u'guarded', 'guard', 'VBN']
    guard -> [u'guard', '

humour: 199 of 472
hunt: 200 of 472
    hunt -> [u'hunt', 'hunt', 'NN']
    hunters -> [u'hunters', 'hunt', 'NNS']
    hunting -> [u'hunting', 'hunt', 'VBG']
    hunt -> [u'hunt', 'hunt', 'NN']
    hunters -> [u'hunters', 'hunt', 'NNS']
    hunt -> [u'hunt', 'hunt', 'NN']
    hunting -> [u'hunting', 'hunt', 'VBG']
    hunter -> [u'hunter', 'hunt', 'NN']
    hunted -> [u'hunted', 'hunt', 'VBN']
    hunts -> [u'hunts', 'hunt', 'NNS']
    hunting -> [u'hunting', 'hunt', 'VBG']
    huntable -> [u'huntable', 'hunt', 'JJ']
    hunts -> [u'hunts', 'hunt', 'NNS']
    hunters -> [u'hunters', 'hunt', 'NNS']
    hunts -> [u'hunts', 'hunt', 'NNS']
    hunted -> [u'hunted', 'hunt', 'VBN']
hunt
hurry: 201 of 472
    hurries -> [u'hurries', 'hurry', 'NNS']
    hurry -> [u'hurry', 'hurry', 'NN']
    hurriedness -> [u'hurriedness', 'hurry', 'NN']
    hurriedly -> [u'hurriedly', 'hurry', 'RB']
    hurriedly -> [u'hurriedly', 'hurry', 'RB']
    hurrying -> [u'hurrying', 'hurry', 'VBG']
    hurry -> [u'hu

intelligent: 213 of 472
    intelligently -> [u'intelligently', 'intelligent', 'RB']
    intelligences -> [u'intelligences', 'intelligent', 'NNS']
    intelligibly -> [u'intelligibly', 'intelligent', 'RB']
    intelligible -> [u'intelligible', 'intelligent', 'JJ']
    intelligent -> [u'intelligent', 'intelligent', 'NN']
    intelligibility -> [u'intelligibility', 'intelligent', 'NN']
    intelligent -> [u'intelligent', 'intelligent', 'NN']
    intelligence -> [u'intelligence', 'intelligent', 'NN']
    intelligibility -> [u'intelligibility', 'intelligent', 'NN']
    intelligently -> [u'intelligently', 'intelligent', 'RB']
    intelligent -> [u'intelligent', 'intelligent', 'NN']
    intelligence -> [u'intelligence', 'intelligent', 'NN']
intelligent
internet: 214 of 472
    internets -> [u'internets', 'internet', 'NNS']
    internet -> [u'internet', 'internet', 'NN']
    internet -> [u'internet', 'internet', 'NN']
    internet -> [u'internet', 'internet', 'NN']
    internets -> [u'interne

kitchen: 232 of 472
    kitchens -> [u'kitchens', 'kitchen', 'NNS']
    kitchens -> [u'kitchens', 'kitchen', 'NNS']
    kitch -> [u'kitch', 'kitchen', 'NN']
    kitchen -> [u'kitchen', 'kitchen', 'NN']
    kitchens -> [u'kitchens', 'kitchen', 'NNS']
    kitchen -> [u'kitchen', 'kitchen', 'NN']
    kitchen -> [u'kitchen', 'kitchen', 'NN']
kitchen
knock: 233 of 472
    knocked -> [u'knocked', 'knock', 'VBN']
    knockers -> [u'knockers', 'knock', 'NNS']
    knock -> [u'knock', 'knock', 'NN']
    knockings -> [u'knockings', 'knock', 'NNS']
    knock -> [u'knock', 'knock', 'NN']
    knocks -> [u'knocks', 'knock', 'NNS']
    knocking -> [u'knocking', 'knock', 'VBG']
    knock -> [u'knock', 'knock', 'NN']
    knocking -> [u'knocking', 'knock', 'VBG']
    knocks -> [u'knocks', 'knock', 'NNS']
    knocker -> [u'knocker', 'knock', 'NN']
    knocked -> [u'knocked', 'knock', 'VBN']
knock
know: 234 of 472
    knowing -> [u'knowing', 'know', 'VBG']
    knows -> [u'knows', 'know', 'NNS']
    know ->

lid: 247 of 472
    lidding -> [u'lidding', 'lid', 'VBG']
    lids -> [u'lids', 'lid', 'NNS']
    lid -> [u'lid', 'lid', 'NN']
    lids -> [u'lids', 'lid', 'NNS']
    lid -> [u'lid', 'lid', 'NN']
    lidded -> [u'lidded', 'lid', 'VBD']
lid
life: 248 of 472
    lifes -> [u'lifes', 'life', 'NN']
    lifers -> [u'lifers', 'life', 'NNS']
    lifer -> [u'lifer', 'life', 'NN']
    life -> [u'life', 'life', 'NN']
    lif -> [u'lif', 'life', 'NN']
    life -> [u'life', 'life', 'NN']
    lifers -> [u'lifers', 'life', 'NNS']
    lifer -> [u'lifer', 'life', 'NN']
    life -> [u'life', 'life', 'NN']
    life -> [u'life', 'life', 'NN']
life
lime: 249 of 472
    lima -> [u'lima', 'lime', 'NN']
    lime -> [u'lime', 'lime', 'NN']
    limes -> [u'limes', 'lime', 'NNS']
    limed -> [u'limed', 'lime', 'VBN']
    liming -> [u'liming', 'lime', 'VBG']
    limes -> [u'limes', 'lime', 'NNS']
    lime -> [u'lime', 'lime', 'NN']
lime
link: 250 of 472
    linkages -> [u'linkages', 'link', 'NNS']
    links -> [

manner: 266 of 472
    manner -> [u'manner', 'manner', 'NN']
    managerially -> [u'managerially', 'manner', 'RB']
    manically -> [u'manically', 'manner', 'RB']
    manfully -> [u'manfully', 'manner', 'RB']
    managerialist -> [u'managerialist', 'manner', 'NN']
    manor -> [u'manor', 'manner', 'NN']
    manners -> [u'manners', 'manner', 'NNS']
    mannerable -> [u'mannerable', 'manner', 'JJ']
    mannerisms -> [u'mannerisms', 'manner', 'NNS']
    mannered -> [u'mannered', 'manner', 'VBN']
    mannerism -> [u'mannerism', 'manner', 'NN']
    mannerist -> [u'mannerist', 'manner', 'NN']
    manner -> [u'manner', 'manner', 'NN']
    mannerly -> [u'mannerly', 'manner', 'RB']
manner
many: 267 of 472
    many -> [u'many', 'many', 'JJ']
    manior -> [u'manior', 'many', 'NN']
    many -> [u'many', 'many', 'JJ']
    many -> [u'many', 'many', 'JJ']
many
marble: 268 of 472
    marbling -> [u'marbling', 'marble', 'VBG']
    marbles -> [u'marbles', 'marble', 'NNS']
    marbleizing -> [u'marbleiz

modest: 285 of 472
    modest -> [u'modest', 'modest', 'JJ']
    modest -> [u'modest', 'modest', 'JJ']
    modest -> [u'modest', 'modest', 'JJ']
    modestly -> [u'modestly', 'modest', 'RB']
    modestly -> [u'modestly', 'modest', 'RB']
modest
mole: 286 of 472
    mol -> [u'mol', 'mole', 'NN']
    moles -> [u'moles', 'mole', 'NNS']
    molar -> [u'molar', 'mole', 'NN']
    mole -> [u'mole', 'mole', 'NN']
    mollies -> [u'mollies', 'mole', 'NNS']
    moll -> [u'moll', 'mole', 'NN']
    mole -> [u'mole', 'mole', 'NN']
    molars -> [u'molars', 'mole', 'NNS']
    molar -> [u'molar', 'mole', 'NN']
    molly -> [u'molly', 'mole', 'RB']
    mole -> [u'mole', 'mole', 'NN']
    moles -> [u'moles', 'mole', 'NNS']
    moles -> [u'moles', 'mole', 'NNS']
    mollis -> [u'mollis', 'mole', 'NNS']
mole
monitor: 287 of 472
    monitoring -> [u'monitoring', 'monitor', 'NN']
    monitory -> [u'monitory', 'monitor', 'NN']
    monitors -> [u'monitors', 'monitor', 'NNS']
    monitors -> [u'monitors', 'mon

network: 300 of 472
    networkability -> [u'networkability', 'network', 'NN']
    networked -> [u'networked', 'network', 'NNS']
    networker -> [u'networker', 'network', 'NN']
    networks -> [u'networks', 'network', 'NNS']
    network -> [u'network', 'network', 'NN']
    networking -> [u'networking', 'network', 'NN']
    networkable -> [u'networkable', 'network', 'JJ']
    networks -> [u'networks', 'network', 'NNS']
    networking -> [u'networking', 'network', 'NN']
    networks -> [u'networks', 'network', 'NNS']
    network -> [u'network', 'network', 'NN']
    network -> [u'network', 'network', 'NN']
    network -> [u'network', 'network', 'NN']
    networking -> [u'networking', 'network', 'NN']
    network -> [u'network', 'network', 'NN']
    networkers -> [u'networkers', 'network', 'NNS']
    networks -> [u'networks', 'network', 'NNS']
    networks -> [u'networks', 'network', 'NNS']
network
next: 301 of 472
    next -> [u'next', 'next', 'JJ']
    nexts -> [u'nexts', 'next', 'NN']


passage: 321 of 472
    passed -> [u'passed', 'passage', 'VBN']
    passed -> [u'passed', 'passage', 'VBN']
    pass -> [u'pass', 'passage', 'NN']
    passage -> [u'passage', 'passage', 'NN']
    passes -> [u'passes', 'passage', 'NNS']
    passed -> [u'passed', 'passage', 'VBN']
    passage -> [u'passage', 'passage', 'NN']
    pass -> [u'pass', 'passage', 'NN']
    passing -> [u'passing', 'passage', 'NN']
    passes -> [u'passes', 'passage', 'NNS']
    passages -> [u'passages', 'passage', 'NNS']
    passages -> [u'passages', 'passage', 'NNS']
    passing -> [u'passing', 'passage', 'NN']
passage
patience: 322 of 472
    patience -> [u'patience', 'patience', 'NN']
    patiently -> [u'patiently', 'patience', 'RB']
    patience -> [u'patience', 'patience', 'NN']
    patiently -> [u'patiently', 'patience', 'RB']
    patience -> [u'patience', 'patience', 'NN']
patience
pc: 323 of 472
    pc -> [u'pc', 'pc', 'NN']
    pc -> [u'pc', 'pc', 'NN']
pc
pedal: 324 of 472
    pedalers -> [u'pedalers'

prick: 338 of 472
    prickliness -> [u'prickliness', 'prick', 'NN']
    pricks -> [u'pricks', 'prick', 'NNS']
    prick -> [u'prick', 'prick', 'NN']
    pricker -> [u'pricker', 'prick', 'NN']
    pricked -> [u'pricked', 'prick', 'NNS']
    prickly -> [u'prickly', 'prick', 'NN']
    pricklier -> [u'pricklier', 'prick', 'NN']
    pricking -> [u'pricking', 'prick', 'VBG']
    pricked -> [u'pricked', 'prick', 'NNS']
    prick -> [u'prick', 'prick', 'NN']
    pricks -> [u'pricks', 'prick', 'NNS']
    pricking -> [u'pricking', 'prick', 'VBG']
prick
priest: 339 of 472
    priests -> [u'priests', 'priest', 'NNS']
    priesthood -> [u'priesthood', 'priest', 'NN']
    priest -> [u'priest', 'priest', 'NN']
    priests -> [u'priests', 'priest', 'NNS']
    priests -> [u'priests', 'priest', 'NNS']
    priesthood -> [u'priesthood', 'priest', 'NN']
    priestly -> [u'priestly', 'priest', 'RB']
    priesthoods -> [u'priesthoods', 'priest', 'NNS']
    priest -> [u'priest', 'priest', 'NN']
    priestly 

readme: 351 of 472
    readme -> [u'readme', 'readme', 'NN']
    readmore -> [u'readmore', 'readme', 'NN']
    readme -> [u'readme', 'readme', 'NN']
    readme -> [u'readme', 'readme', 'NN']
    readme -> [u'readme', 'readme', 'NN']
readme
real: 352 of 472
    real -> [u'real', 'real', 'JJ']
    realistic -> [u'realistic', 'real', 'JJ']
    reallly -> [u'reallly', 'real', 'RB']
    reallity -> [u'reallity', 'real', 'NN']
    realizing -> [u'realizing', 'real', 'VBG']
    real -> [u'real', 'real', 'JJ']
    realized -> [u'realized', 'real', 'VBN']
    realizes -> [u'realizes', 'real', 'NNS']
    really -> [u'really', 'real', 'RB']
    real -> [u'real', 'real', 'JJ']
    realistically -> [u'realistically', 'real', 'RB']
    reall -> [u'reall', 'real', 'NN']
    reality -> [u'reality', 'real', 'NN']
    realism -> [u'realism', 'real', 'NN']
    realities -> [u'realities', 'real', 'NNS']
    realization -> [u'realization', 'real', 'NN']
    realize -> [u'realize', 'real', 'VB']
    really 

row: 365 of 472
    rowing -> [u'rowing', 'row', 'VBG']
    rowers -> [u'rowers', 'row', 'NNS']
    row -> [u'row', 'row', 'NN']
    rower -> [u'rower', 'row', 'NN']
    rows -> [u'rows', 'row', 'NNS']
    row -> [u'row', 'row', 'NN']
    rower -> [u'rower', 'row', 'NN']
    rowed -> [u'rowed', 'row', 'NN']
    rowed -> [u'rowed', 'row', 'NN']
    rows -> [u'rows', 'row', 'NNS']
    row -> [u'row', 'row', 'NN']
row
rude: 366 of 472
    rude -> [u'rude', 'rude', 'NN']
    rudeness -> [u'rudeness', 'rude', 'NN']
    rude -> [u'rude', 'rude', 'NN']
    rudely -> [u'rudely', 'rude', 'RB']
    rude -> [u'rude', 'rude', 'NN']
    rudeness -> [u'rudeness', 'rude', 'NN']
    rudely -> [u'rudely', 'rude', 'RB']
    ruder -> [u'ruder', 'rude', 'NN']
rude
rust: 367 of 472
    rust -> [u'rust', 'rust', 'NN']
    rusted -> [u'rusted', 'rust', 'VBN']
    rusting -> [u'rusting', 'rust', 'VBG']
    rusted -> [u'rusted', 'rust', 'VBN']
    rusts -> [u'rusts', 'rust', 'NNS']
    rust -> [u'rust', 'rust'

shoulder: 381 of 472
    shoulders -> [u'shoulders', 'shoulder', 'NNS']
    shoulder -> [u'shoulder', 'shoulder', 'NN']
    shoulders -> [u'shoulders', 'shoulder', 'NNS']
    shoulder -> [u'shoulder', 'shoulder', 'NN']
    shoulder -> [u'shoulder', 'shoulder', 'NN']
    shouldering -> [u'shouldering', 'shoulder', 'VBG']
    shouldering -> [u'shouldering', 'shoulder', 'VBG']
    shouldered -> [u'shouldered', 'shoulder', 'VBN']
shoulder
show: 382 of 472
    showing -> [u'showing', 'show', 'VBG']
    show -> [u'show', 'show', 'NN']
    showing -> [u'showing', 'show', 'VBG']
    showed -> [u'showed', 'show', 'VBD']
    showings -> [u'showings', 'show', 'NNS']
    show -> [u'show', 'show', 'NN']
    show -> [u'show', 'show', 'NN']
    showed -> [u'showed', 'show', 'VBD']
    showes -> [u'showes', 'show', 'NN']
    shows -> [u'shows', 'show', 'NNS']
    shows -> [u'shows', 'show', 'NNS']
    shows -> [u'shows', 'show', 'NNS']
show
shutdown: 383 of 472
    shutdown -> [u'shutdown', 'shutdown'

stair: 397 of 472
    stairs -> [u'stairs', 'stair', 'NNS']
    stair -> [u'stair', 'stair', 'NN']
    stair -> [u'stair', 'stair', 'NN']
    stairs -> [u'stairs', 'stair', 'NNS']
stair
startup: 398 of 472
    startup -> [u'startup', 'startup', 'NN']
    startup -> [u'startup', 'startup', 'NN']
    startup -> [u'startup', 'startup', 'NN']
    startup -> [u'startup', 'startup', 'NN']
    startups -> [u'startups', 'startup', 'NNS']
    startups -> [u'startups', 'startup', 'NNS']
startup
steady: 399 of 472
    steadied -> [u'steadied', 'steady', 'VBN']
    steadies -> [u'steadies', 'steady', 'NNS']
    steadier -> [u'steadier', 'steady', 'NN']
    steady -> [u'steady', 'steady', 'JJ']
    steadily -> [u'steadily', 'steady', 'RB']
    steadying -> [u'steadying', 'steady', 'VBG']
    steady -> [u'steady', 'steady', 'JJ']
    steadied -> [u'steadied', 'steady', 'VBN']
    steadies -> [u'steadies', 'steady', 'NNS']
    steadier -> [u'steadier', 'steady', 'NN']
    steadying -> [u'steadying', 

tap: 419 of 472
    tapping -> [u'tapping', 'tap', 'VBG']
    tapping -> [u'tapping', 'tap', 'VBG']
    tappings -> [u'tappings', 'tap', 'NNS']
    tapped -> [u'tapped', 'tap', 'NNS']
    tappable -> [u'tappable', 'tap', 'JJ']
    tap -> [u'tap', 'tap', 'NN']
    taps -> [u'taps', 'tap', 'NNS']
    tapper -> [u'tapper', 'tap', 'NN']
    tapped -> [u'tapped', 'tap', 'NNS']
    tap -> [u'tap', 'tap', 'NN']
    tappers -> [u'tappers', 'tap', 'NNS']
    tappers -> [u'tappers', 'tap', 'NNS']
tap
tear: 420 of 472
    tear -> [u'tear', 'tear', 'NN']
    tears -> [u'tears', 'tear', 'NNS']
    tear -> [u'tear', 'tear', 'NN']
    tears -> [u'tears', 'tear', 'NNS']
    teared -> [u'teared', 'tear', 'VBN']
    tearer -> [u'tearer', 'tear', 'NN']
    tearing -> [u'tearing', 'tear', 'VBG']
    tearing -> [u'tearing', 'tear', 'VBG']
tear
tell: 421 of 472
    telling -> [u'telling', 'tell', 'VBG']
    telling -> [u'telling', 'tell', 'VBG']
    telling -> [u'telling', 'tell', 'VBG']
    tells -> [u'tel

truck: 438 of 472
    trucked -> [u'trucked', 'truck', 'NNS']
    trucker -> [u'trucker', 'truck', 'NN']
    trucking -> [u'trucking', 'truck', 'NN']
    truck -> [u'truck', 'truck', 'NN']
    truckful -> [u'truckful', 'truck', 'NN']
    truck -> [u'truck', 'truck', 'NN']
    trucks -> [u'trucks', 'truck', 'NNS']
    trucker -> [u'trucker', 'truck', 'NN']
    truckish -> [u'truckish', 'truck', 'NN']
    trucks -> [u'trucks', 'truck', 'NNS']
    truckers -> [u'truckers', 'truck', 'NNS']
    trucking -> [u'trucking', 'truck', 'NN']
    truckers -> [u'truckers', 'truck', 'NNS']
    trucking -> [u'trucking', 'truck', 'NN']
    trucked -> [u'trucked', 'truck', 'NNS']
    trucker -> [u'trucker', 'truck', 'NN']
    truck -> [u'truck', 'truck', 'NN']
    trucks -> [u'trucks', 'truck', 'NNS']
truck
try: 439 of 472
    trying -> [u'trying', 'try', 'VBG']
    try -> [u'try', 'try', 'NN']
    trys -> [u'trys', 'try', 'NN']
    try -> [u'try', 'try', 'NN']
    try -> [u'try', 'try', 'NN']
    tryin

vegetable: 452 of 472
    vegetatively -> [u'vegetatively', 'vegetable', 'RB']
    vegetarians -> [u'vegetarians', 'vegetable', 'NNS']
    vegetation -> [u'vegetation', 'vegetable', 'NN']
    vegeterian -> [u'vegeterian', 'vegetable', 'JJ']
    vegetarianism -> [u'vegetarianism', 'vegetable', 'NN']
    vegetarian -> [u'vegetarian', 'vegetable', 'JJ']
    vegetarian -> [u'vegetarian', 'vegetable', 'JJ']
    vegetables -> [u'vegetables', 'vegetable', 'NNS']
    vegetarianism -> [u'vegetarianism', 'vegetable', 'NN']
    vegetations -> [u'vegetations', 'vegetable', 'NNS']
    vegetables -> [u'vegetables', 'vegetable', 'NNS']
    vegetable -> [u'vegetable', 'vegetable', 'NN']
    vegetarians -> [u'vegetarians', 'vegetable', 'NNS']
    vegetable -> [u'vegetable', 'vegetable', 'NN']
    vegetable -> [u'vegetable', 'vegetable', 'NN']
    vegetal -> [u'vegetal', 'vegetable', 'NN']
    vegetarian -> [u'vegetarian', 'vegetable', 'JJ']
    vegetables -> [u'vegetables', 'vegetable', 'NNS']
vegetabl

whether: 463 of 472
    whether -> [u'whether', 'whether', 'IN']
    whether -> [u'whether', 'whether', 'IN']
    whether -> [u'whether', 'whether', 'IN']
    whethere -> [u'whethere', 'whether', 'RB']
whether
widow: 464 of 472
    widow -> [u'widow', 'widow', 'NN']
    widows -> [u'widows', 'widow', 'NNS']
    widowhood -> [u'widowhood', 'widow', 'NN']
    widower -> [u'widower', 'widow', 'NN']
    widowhood -> [u'widowhood', 'widow', 'NN']
    widowed -> [u'widowed', 'widow', 'NN']
    widows -> [u'widows', 'widow', 'NNS']
    widowers -> [u'widowers', 'widow', 'NNS']
    widows -> [u'widows', 'widow', 'NNS']
    widow -> [u'widow', 'widow', 'NN']
    widowed -> [u'widowed', 'widow', 'NN']
    widower -> [u'widower', 'widow', 'NN']
    widow -> [u'widow', 'widow', 'NN']
    widowers -> [u'widowers', 'widow', 'NNS']
    widowed -> [u'widowed', 'widow', 'NN']
    widower -> [u'widower', 'widow', 'NN']
widow
wife: 465 of 472
    wifed -> [u'wifed', 'wife', 'NN']
    wife -> [u'wife', 'w

## Look for Words I Need to Add

Using all articles linked to the front page of the [simple english wikipedia](https://simple.wikipedia.org/wiki/Main_Page), or at least the knowledge groups, I can look for certain words that come up a lot and add these to my dictionary as needed. 

In [151]:
# get my sentences
sentences = pickle.load(open('../data/sentences.pickle', "rb" ))

In [174]:
import time
# Import the built-in logging module and configure it so that Word2Vec 
# creates nice output messages

import logging
logging.basicConfig(format='%(asctime)s : %(levelname)s : %(message)s',\
    level=logging.INFO)

start = time.clock()
# Set values for various parameters
num_features = 300    # Word vector dimensionality                      
min_word_count = 40   # Minimum word count                        
num_workers = 4       # Number of threads to run in parallel
context = 10          # Context window size                                                                                    
downsampling = 1e-3   # Downsample setting for frequent words

# Initialize and train the model (this will take some time)
from gensim.models import word2vec

model_name = 'all_parsed'+str(num_features)+'features_'+str(min_word_count)+'min_word_count_'+str(context)+'context.npy'

print "Training model..."
amodel = word2vec.Word2Vec(sentences, workers=num_workers, 
                          size=num_features, min_count = min_word_count, 
                          window = context, sample = downsampling)

# If you don't plan to train the model any further, calling 
# init_sims will make the model much more memory-efficient.
# model.init_sims(replace=True)

# It can be helpful to create a meaningful model name and 
# save the model for later use. You can load it later using Word2Vec.load()

amodel.save('../data/'+model_name)
print '{:.2f}s'.format(time.clock() - start)
print len(amodel.wv.vocab.keys())

2017-06-30 13:45:28,490 : INFO : collecting all words and their counts
2017-06-30 13:45:28,492 : INFO : PROGRESS: at sentence #0, processed 0 words, keeping 0 word types
2017-06-30 13:45:28,532 : INFO : collected 11387 word types from a corpus of 153805 raw words and 9987 sentences
2017-06-30 13:45:28,535 : INFO : Loading a fresh vocabulary
2017-06-30 13:45:28,548 : INFO : min_count=40 retains 436 unique words (3% of original 11387, drops 10951)
2017-06-30 13:45:28,549 : INFO : min_count=40 leaves 96729 word corpus (62% of original 153805, drops 57076)
2017-06-30 13:45:28,556 : INFO : deleting the raw counts dictionary of 11387 items
2017-06-30 13:45:28,562 : INFO : sample=0.001 downsamples 61 most-common words
2017-06-30 13:45:28,565 : INFO : downsampling leaves estimated 53689 word corpus (55.5% of prior 96729)
2017-06-30 13:45:28,568 : INFO : estimated required memory for 436 words and 300 dimensions: 1264400 bytes
2017-06-30 13:45:28,573 : INFO : resetting layer weights
2017-06-30 

Training model...


2017-06-30 13:45:28,980 : INFO : worker thread finished; awaiting finish of 3 more threads
2017-06-30 13:45:28,989 : INFO : worker thread finished; awaiting finish of 2 more threads
2017-06-30 13:45:28,993 : INFO : worker thread finished; awaiting finish of 1 more threads
2017-06-30 13:45:28,997 : INFO : worker thread finished; awaiting finish of 0 more threads
2017-06-30 13:45:29,000 : INFO : training on 769025 raw words (268899 effective words) took 0.4s, 671803 effective words/s
2017-06-30 13:45:29,006 : INFO : saving Word2Vec object under ../data/all_parsed300features_40min_word_count_10context.npy, separately None
2017-06-30 13:45:29,010 : INFO : not storing attribute syn0norm
2017-06-30 13:45:29,015 : INFO : not storing attribute cum_table
2017-06-30 13:45:29,029 : INFO : saved ../data/all_parsed300features_40min_word_count_10context.npy


0.54s
436


In [175]:
 lst = ['CD', # CD: numeral, cardinal
         'EX', # EX: existential there
         'FW', # FW: foreign word
         'LS', # LS: list item marker
         'NNP', # NNP: noun, proper, singular
         'NNPS', # NNPS: noun, proper, plural
         'PRP', # PRP: pronoun, personal
         'SYM', # SYM: symbol
         'TO', # TO: "to" as preposition or infinitive marker
         'POS',
         'JJ',
         '$', # $: dollar
         '(',
         ')',
         ',',
         '.',
         ':',
         '"'
        ] 

In [180]:
from nltk import pos_tag
import string

d = pickle.load(open('../data/basic_english.pickle', 'rb'))
dkeys = d.keys()
n = 0
new_words = []
print 'add_lst = []'
for key in amodel.wv.vocab.keys():
    k = key.strip(string.punctuation).lower()
    if k not in dkeys and len(k) >= 1:
        tag = pos_tag([k])[0][1]
        if tag not in lst:
            new_words.append(k)
new_words = list(set(new_words))
new_words.sort()
for word in new_words:
    print "add_lst.append('{}')".format(word)
        


add_lst = []
add_lst.append('and')
add_lst.append('displaystyle')
add_lst.append('of')


Already added (preserved for science):


add_lst = []
add_lst.append('accord')
add_lst.append('algebra')
add_lst.append('although')
add_lst.append('and')
add_lst.append('another')
add_lst.append('anthropology')
add_lst.append('around')
add_lst.append('assembly')
add_lst.append('astronomy')
add_lst.append('atheism')
add_lst.append('atom')
add_lst.append('b')
add_lst.append('become')
add_lst.append('been')
add_lst.append('began')
add_lst.append('be')
add_lst.append('best')
add_lst.append('buddha')
add_lst.append('build')
add_lst.append('come')
add_lst.append('cannot')
add_lst.append('catholic')
add_lst.append('century')
add_lst.append('christian')
add_lst.append('circuit')
add_lst.append('compose')
add_lst.append('consider')
add_lst.append('constitution')
add_lst.append('copyright')
add_lst.append('create')
add_lst.append('culture')
add_lst.append('dance')
add_lst.append('describe')
add_lst.append('display')
add_lst.append('divide')
add_lst.append('do')
add_lst.append('during')
add_lst.append('ecology')
add_lst.append('energy')
add_lst.append('engineering')
add_lst.append('especially')
add_lst.append('europe')
add_lst.append('everything')
add_lst.append('follow')
add_lst.append('found')
add_lst.append('france')
add_lst.append('game')
add_lst.append('greek')
add_lst.append('had')
add_lst.append('has')
add_lst.append('health')
add_lst.append('hindu')
add_lst.append('holy')
add_lst.append('however')
add_lst.append('ideas')
add_lst.append('include')
add_lst.append('instead')
add_lst.append('jazz')
add_lst.append('jesus')
add_lst.append('jew')
add_lst.append('judaism')
add_lst.append('just')
add_lst.append('king')
add_lst.append('known')
add_lst.append('led')
add_lst.append('less')
add_lst.append('lot')
add_lst.append('made')
add_lst.append('mathematics')
add_lst.append('matter')
add_lst.append('more')
add_lst.append('most')
add_lst.append('movie')
add_lst.append('muslim')
add_lst.append('myth')
add_lst.append('object')
add_lst.append('of')
add_lst.append('often')
add_lst.append('oldest')
add_lst.append('orthodox')
add_lst.append('our')
add_lst.append('period')
add_lst.append('philosophy')
add_lst.append('physics')
add_lst.append('psychology')
add_lst.append('radar')
add_lst.append('radio')
add_lst.append('research')
add_lst.append('roman')
add_lst.append('said')
add_lst.append('sociology')
add_lst.append('someone')
add_lst.append('something')
add_lst.append('sometime')
add_lst.append('standard')
add_lst.append('state')
add_lst.append('style')
add_lst.append('taught')
add_lst.append('telescope')
add_lst.append('television')
add_lst.append('term')
add_lst.append('these')
add_lst.append('those')
add_lst.append('though')
add_lst.append('tool')
add_lst.append('torah')
add_lst.append('until')
add_lst.append('usual')
add_lst.append('which')
add_lst.append('within')
add_lst.append('without')
add_lst.append('worship')
add_lst.append('wrote')
add_lst.append('x')
add_lst.append('y')

add_lst = []
add_lst.append('according')
add_lst.append('and')
add_lst.append('became')
add_lst.append('being')
add_lst.append('built')
add_lst.append('came')
add_lst.append('christianity')
add_lst.append('christians')
add_lst.append('de')
add_lst.append('displaystyle')
add_lst.append('does')
add_lst.append('done')
add_lst.append('hindus')
add_lst.append('of')
add_lst.append('sometimes')

In [179]:
my_dict = make_dictionary(add_lst)

Running...
15


C:\Users\darin\Anaconda2\lib\site-packages\ipykernel_launcher.py:16: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.
  app.launch_new_instance()


starting...
according: 0 of 15
    according -> [u'according', 'according', 'VBG']
    according -> [u'according', 'according', 'VBG']
    accordance -> [u'accordance', 'according', 'NN']
    according -> [u'according', 'according', 'VBG']
according
and: 1 of 15
became: 2 of 15
    became -> [u'became', 'became', 'VBD']
    became -> [u'became', 'became', 'VBD']
    becames -> [u'becames', 'became', 'NNS']
became
being: 3 of 15
    being -> [u'being', 'being', 'VBG']
    being -> [u'being', 'being', 'VBG']
    being -> [u'being', 'being', 'VBG']
being
built: 4 of 15
    built -> [u'built', 'built', 'NN']
    builts -> [u'builts', 'built', 'NNS']
    built -> [u'built', 'built', 'NN']
    built -> [u'built', 'built', 'NN']
built
came: 5 of 15
    came -> [u'came', 'came', 'VBD']
    cames -> [u'cames', 'came', 'NNS']
    cameran -> [u'cameran', 'came', 'NN']
    came -> [u'came', 'came', 'VBD']
came
christianity: 6 of 15
    christianity -> [u'christianity', 'christianity', 'NN']
    ch